src/meta/src/hummock/compaction/compaction_picker.rs

```diff
@@ -189,13 +190,131 @@ impl CompactionPicker for MinOverlappingPicker {
     }
 }
 
+pub struct ManualCompactionPicker {
+    compact_task_id: u64,
+    overlap_strategy: Arc<dyn OverlapStrategy>,
+    option: ManualCompactionOption,
+}
+
+impl ManualCompactionPicker {
+    pub fn new(
+        compact_task_id: u64,
+        overlap_strategy: Arc<dyn OverlapStrategy>,
+        option: ManualCompactionOption,
+    ) -> Self {
+        Self {
+            compact_task_id,
+            overlap_strategy,
+            option,
+        }
+    }
+}
+
+impl CompactionPicker for ManualCompactionPicker {
+    fn pick_compaction(
+        &self,
+        levels: &[Level],
+        level_handlers: &mut [LevelHandler],
+    ) -> Option<SearchResult> {
+        let level = self.option.level;
+        let target_level = level + 1;
```

Comment:
Could `self.option.level` be zero?
If `self.option.level` is zero, we shall compact data to `base_level` rather than `level + 1`

Reply:
self.option.level will be zero.  shall we need a flag to control the intra L0 compaction ?